gradle-incremental-configuration-cache/src/main/java/com/palantir/gradle/configcache/incremental/DryRunConfigurationCacheEnabledTask.java

```diff
@@ -0,0 +1,210 @@
+import com.palantir.gradle.configcache.incremental.DryRunResult.Success;
+import com.palantir.gradle.utils.circleciartifacts.ArtifactLocation;
+import com.palantir.gradle.utils.circleciartifacts.CircleCiArtifacts;
+import com.palantir.gradle.utils.environmentvariables.EnvironmentVariables;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.TaskAction;
+
+public abstract class DryRunConfigurationCacheEnabledTask extends AbstractDryRunTask {
+
+    @Nested
+    protected abstract EnvironmentVariables getEnvironmentVariables();
+
+    @Nested
+    protected abstract CircleCiArtifacts getCircleCiArtifacts();
+
+    public DryRunConfigurationCacheEnabledTask() {
+        getMarkerOutputFile()
+                .set(getTemporaryDir()
+                        .toPath()
+                        .resolve("dryRunConfigurationCacheEnabled.marker")
+                        .toFile());
+    }
+
+    @TaskAction
+    public final void check() throws IOException {
+        DryRunResult result =
+                dryRun(List.of("--configuration-cache", "-Pconfiguration-cache-compatible-for-all-tasks"));
+
+        if (result instanceof Success) {
+            Files.writeString(getMarkerOutputFile().get().getAsFile().toPath(), "up-to-date");
+            return;
+        }
+
+        String message = errorMessage(((Failure) result).errorOutput());
+        throw new RuntimeException(message);
+    }
+
+    private String errorMessage(String output) {
+        boolean isCircleCI = getEnvironmentVariables()
+                .envVarOrFromTestingProperty("CIRCLECI")
+                .isPresent();
+
+        if (!isCircleCI) {
+            return buildDetailedErrorMessage(output, Optional.empty());
+        }
+
+        return circleCiErrorMessage(output);
+    }
+
+    private String circleCiErrorMessage(String output) {
+        try {
+            String artifactPath = "configuration-cache-validation-report/validation-report.txt";
+            ArtifactLocation artifactLocation =
+                    getCircleCiArtifacts().resolveArtifactLocation(artifactPath).get();
+
+            Path artifactLocationPath =
+                    artifactLocation.physicalPath().getAsFile().toPath();
+            Files.createDirectories(artifactLocationPath.getParent());
+            Files.writeString(artifactLocationPath, output);
+
+            return buildDetailedErrorMessage(output, Optional.of(artifactLocation.circleLink()));
+        } catch (IOException e) {
+            // Fall back to including output directly if we can't create the artifact
+            return String.format(
+                    """
+                    %s
+
+                    (Failed to save CircleCI artifact: %s)
+                    """,
+                    buildDetailedErrorMessage(output, Optional.empty()), e.getMessage());
```

Comment:
Rather than `e.getMessage()`, we should print out the stacktrace. You can use `Throwables.getStackTraceAsString` if you have guava, otherwise we can just copy the method:

```
public static String getStackTraceAsString(Throwable throwable) {
    StringWriter stringWriter = new StringWriter();
    throwable.printStackTrace(new PrintWriter(stringWriter));
    return stringWriter.toString();
  }
```